File: src/main/java/info/dddeurope/lab/app/repositories/UserRepository.java
package info.dddeurope.lab.app.repositories;

import java.util.HashMap;
import java.util.Map;

import info.dddeurope.lab.app.domain.UserAggregate.User;

/*

    Here we use a simple in-meomory key-map value to represent a document-based database.
    In real-world, you would instead use a real database, such as MongoDB.

*/

public class UserRepository {

    Map<String, User> users;
    
    public UserRepository() {
        this.users = new HashMap<>();
    }

    public void set(String userId, User user) {
        this.users.put(userId, user);
    }

    public User get(String userId) {
        return this.users.get(userId);
    }

}

File: DotNet/App/Repositories/UserRepository.cs
using System.Collections.Generic;
using App.Domain.UserAggregate;

namespace App.Repositories
{
    /*
        Here we use a simple in-memory key-map value to represent a document-based database.
        In real-world, you would instead use a real database, such as MongoDB.
    */

    public class UserRepository
    {
        readonly Dictionary<string, User> _users;

        public UserRepository()
        {
            _users = new Dictionary<string, User>();
        }

        public void Set(string userId, User user)
        {
            _users[userId] = user;
        }

        public User Get(string userId)
        {
            return _users.TryGetValue(userId, out var value) ? value : null;
        }
    }
}
